### PR TITLE
feat: self deleting messages group settings [AR-3226]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- set message self deleting for group in group settings
- disable self deleting messages option for user where group enforces this time
- handle self deleting message group event
- added migration for both message and conversation table to support message about changed timer for group and message timer in group conversation table

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt [hunk 1]
```diff
@@ -302,6 +302,23 @@ sealed class Event(open val id: String, open val transient: Boolean) {
                 senderUserIdKey to senderUserId.toLogString(),
             )
         }
+
+        data class ConversationMessageTimer(
+            override val id: String,
+            override val conversationId: ConversationId,
+            override val transient: Boolean,
+            val messageTimer: Long?,
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt:310]: Why don't we declare value as `Duration` type for coherence with the other duration/timeouts values?
```suggestion
            val messageTimerDuration: Duration,
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt [hunk 2]
```diff
@@ -47,6 +48,7 @@ data class Conversation(
     val accessRole: List<AccessRole>,
     val creatorId: String?,
     val receiptMode: ReceiptMode,
+    val messageTimer: Duration? // enforced time by group conversation for self deleting messages, always NULL for 1on1 conversation
```
Comment (trOnk12) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt:51]: I think messageTimer does not explain the purpose of this well enough

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationGroupRepository.kt [hunk 3]
```diff
@@ -271,4 +272,11 @@ internal class ConversationGroupRepositoryImpl(
 
     override suspend fun observeGuestRoomLink(conversationId: ConversationId): Flow<String?> =
         conversationDAO.observeGuestRoomLinkByConversationId(conversationId.toDao())
+
+    override suspend fun updateMessageTimer(conversationId: ConversationId, messageTimer: Long?): Either<NetworkFailure, Unit> =
```
Comment (trOnk12) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationGroupRepository.kt:276]: I think messageTimer does not explain the purpose of this well enough, for someone outside the context

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt [hunk 4]
```diff
@@ -302,6 +302,23 @@ sealed class Event(open val id: String, open val transient: Boolean) {
                 senderUserIdKey to senderUserId.toLogString(),
             )
         }
+
+        data class ConversationMessageTimer(
```
Comment (trOnk12) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt:306]: I think messageTimer does not explain the purpose of this well enough, for someone outside the context

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt [hunk 5]
```diff
@@ -970,6 +972,12 @@ class UserSessionScope internal constructor(
             persistMessage = persistMessage
         )
 
+    private val conversationMessageTimerEventHandler: ConversationMessageTimerEventHandler
```
Comment (trOnk12) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt:975]: I think messageTimer does not explain the purpose of this well enough, for someone outside the context

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfdeletingMessages/ObserveSelfDeletionTimerSettingsForConversationUseCase.kt [hunk 6]
```diff
@@ -36,41 +36,46 @@ interface ObserveSelfDeletionTimerSettingsForConversationUseCase {
...
-                            teamSettingsTimer
-                        } else {
-                            conversationTimer.selfDeletionTimer
-                        }
-                    })
-                })
-            }
+            .flatMapRightWithEither {
+                when (it.enforcedSelfDeletionTimer) {
+                    SelfDeletionTimer.Disabled -> flowOf(Either.Right(SelfDeletionTimer.Disabled))
```
Comment (trOnk12) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfdeletingMessages/ObserveSelfDeletionTimerSettingsForConversationUseCase.kt:51]: like I siad in previous PR i think selfdeletiontimer is confusing name

**Review by trOnk12 — APPROVED** [review]
LGTM I just tihnk some names could be improved, as for someone outside the context it will hard to know exactly what they mean

**Review by gongracr — APPROVED** [review]
LGTM 🍪 Just make sure the migration action is fixed before merging

File: persistence/src/commonMain/db_user/schemas/36.db
Comment (MohamadJaara) [inline persistence/src/commonMain/db_user/schemas/36.db:1]: is there a reason for updating 36.de and 39.db ?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/ConversationMessageTimerEventHandler.kt [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.PersistMessageUseCase
+import com.wire.kalium.logic.functional.onFailure
+import com.wire.kalium.logic.functional.onSuccess
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.logic.wrapStorageRequest
+import com.wire.kalium.persistence.dao.ConversationDAO
+import com.wire.kalium.util.DateTimeUtil
+
+interface ConversationMessageTimerEventHandler {
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/ConversationMessageTimerEventHandler.kt:36]: We should also cover this class with tests

File: persistence/src/commonMain/db_user/schemas/36.db
Comment (Garzas) [reply persistence/src/commonMain/db_user/schemas/36.db:1]: Removed

**Review by MohamadJaara — APPROVED** [review]

**Pull request merged**